src/epd/Refs.java

```diff
@@ -1,25 +1,197 @@
 package epd;
 
+import epd.io.conversion.DataSetRefExtension;
+import org.openlca.ilcd.commons.IDataSet;
 import org.openlca.ilcd.commons.Ref;
 import org.openlca.ilcd.flows.Flow;
+import org.openlca.ilcd.processes.ComplianceDeclaration;
+import org.openlca.ilcd.processes.Modelling;
+import org.openlca.ilcd.processes.Process;
+import org.openlca.ilcd.processes.Publication;
+import org.openlca.ilcd.processes.Review;
+import org.openlca.ilcd.processes.Technology;
+import org.openlca.ilcd.util.Epds;
+import org.openlca.ilcd.util.Flows;
 import org.openlca.ilcd.util.Processes;
 
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import java.util.Objects;
 
 public final class Refs {
 
 	private Refs() {
 	}
+	
+	public static List<Ref> getAllEditable(IDataSet ds) {
+		if (ds instanceof Process epd)
+			return getAllEditable(epd);
+		else if (ds instanceof Flow product) {
+			return getAllEditable(product);
+		} else return Collections.emptyList();
+	}
+
+	public static List<Ref> getAllEditable(Process epd) {
+		var refs = new ArrayList<Ref>();
+
+		refs.addAll(getAllProcessInfo(epd));
+
+		var inventoryMethod = Processes.getInventoryMethod(epd);
+		if (inventoryMethod != null) {
+			refs.addAll(inventoryMethod.getSources());
+		}
+
+		var representativeness = Processes.getRepresentativeness(epd);
+		if (representativeness != null) {
+			refs.addAll(representativeness.getDataHandlingSources());
+			refs.addAll(representativeness.getSources());
+		}
+
+		refs.addAll(Epds.withOriginalEpds(epd));
 
-	public static List<Ref> getAllEditable(Processes epd) {
-		// TODO:
-		return Collections.emptyList();
+		var modelling = Processes.getModelling(epd);
+		if (modelling != null) {
+			refs.addAll(getAll(modelling));
+		}
+		
+		refs.addAll(getAllAdmin(epd));
+
+		return refs;
 	}
 
+
 	public static List<Ref> getAllEditable(Flow product) {
-		// TODO:
-		return Collections.emptyList();
+		var genericFlow = getGenericFlow(product);
+		return genericFlow != null
+			? Collections.singletonList(genericFlow)
+			: Collections.emptyList();
+	}
+
+	private static List<Ref> getAll(Modelling modelling) {
+		var refs = new ArrayList<Ref>();
+
+		modelling.getComplianceDeclarations().stream()
+			.map(ComplianceDeclaration::getSystem)
+			.filter(Objects::nonNull)
+			.forEach(refs::add);
+
+		var validation = modelling.getValidation();
+		if (validation != null) {
+			validation.getReviews().stream()
+				.map(Review::getReviewers)
+				.forEach(refs::addAll);
+			validation.getReviews().stream()
+				.map(Review::getReport)
+				.filter(Objects::nonNull)
+				.forEach(refs::add);
+		}
+
+		return refs;
+	}
+
+	private static Ref getGenericFlow(Flow product) {
+		var info = Flows.getDataSetInfo(product);
+		if (info == null)
+			return null;
+
+		var extension = info.getEpdExtension();
+		if (extension == null)
+			return null;
+
+		return DataSetRefExtension.readFlow("isA", extension);
```

Comment:
I am not sure this is the way to go

Reply:
no, this field is in the JAXB model now (`genericFlow`)

Reply:
I have pushed a change.